### PR TITLE
automation: Fix Fedora29 dev image

### PR DESCRIPTION
**Pull request by EdDev**
Udev has stopped working in the Fedora dev image.
It seems that the source is related to a missing systemd-udev package.

The container dockerfile has been updated to include systemd-udev.
In addition, an update has been included to confirm with the systemd
container enablement instructions:
https://hub.docker.com/r/fedora/systemd-systemd/

systemd-udevd is also being started when the container is run and NM is
restarted in order to confirm it loads with udev activated already.

### CODE REVIEW

**Comment by tyll** [comment]
Since this indicates a packaging/configuration error in NetworkManager, I filed https://bugzilla.redhat.com/show_bug.cgi?id=1704456

I am not sure about what you mean about updating in regard to the systemd docker recommendations - is it the dnf update?

File: automation/Dockerfile.fedora [hunk 1]
```diff
@@ -6,7 +6,12 @@ FROM fedora
 ENV container docker
 COPY docker_enable_systemd.sh docker_sys_config.sh ./
 
-RUN bash ./docker_enable_systemd.sh && rm ./docker_enable_systemd.sh
+RUN \
+    dnf -y update && \
+    dnf install -y systemd-udev && \
```
Comment (tyll) [inline automation/Dockerfile.fedora]: IMHO this should be added to the NM section in the other dnf command. Also a reference to https://bugzilla.redhat.com/show_bug.cgi?id=1704456 would be great so it is clear that it can be removed eventually.

File: automation/run-tests.sh [hunk 2]
```diff
@@ -258,8 +258,11 @@ then
 fi
 
 docker_exec 'while ! systemctl is-active dbus; do sleep 1; done'
+docker_exec 'systemctl start systemd-udevd
```
Comment (tyll) [inline automation/run-tests.sh]: Is this really needed?

Comment (tyll) [reply automation/run-tests.sh]: Also, if it is needed, the CentOS images should be fixed to require systemd-udevd as well.

File: automation/run-tests.sh [hunk 3]
```diff
@@ -258,8 +258,11 @@ then
...
+             while ! systemctl is-active systemd-udevd; do sleep 1; done
+'
 docker_exec '
-    systemctl start NetworkManager
+    systemctl restart NetworkManager
```
Comment (tyll) [inline automation/run-tests.sh]: Why is this needed?

File: automation/Dockerfile.fedora (hunk 1 above)
Comment (EdDev) [reply automation/Dockerfile.fedora]: I somehow considered the udev, part of systemd patching, that is why I placed it here.
But I guess it is not related, I will move it.

File: automation/run-tests.sh (hunk 2 above)
Comment (EdDev) [reply automation/run-tests.sh]: On CentoOS it seems to be part of the systemd package and it comes enabled.
On Fedora, it was a requirement of `dracut`, but now they seems to have dropped the requirement.

So yes, it is needed otherwise it does not work.

File: automation/run-tests.sh (hunk 3 above)
Comment (EdDev) [reply automation/run-tests.sh]: Based on my tests, the NM service needs to be restarted after systemd-udevd is running, otherwise it does not work. That is why we need to wait for that daemon to be active before we proceed.

File: automation/run-tests.sh (hunk 2 above)
Comment (EdDev) [reply automation/run-tests.sh]: Correction: `dracut` itself got dropped from the base image, that is why `systemd-udev` is now missing, not because of requirement changes in `dracut`.

File: automation/run-tests.sh (hunk 3 above)
Comment (tyll) [reply automation/run-tests.sh]: this would be strange because we did not have to restart NM before and the breaking change was that systemd-udev was not installed anymore.

**Comment by tyll** [comment]
> > Since this indicates a packaging/configuration error in NetworkManager, I filed https://bugzilla.redhat.com/show_bug.cgi?id=1704456
> 
> Ok, but I am not sure if applications should depend on udevd in this manner. Only `dracut` seemed to have depended on it.

NetworkManager depends on udevd, otherwise we would not have this problem. The problem is that the package does not express this dependency properly. Also I am not sure why it would be wrong for an application to depend on udevd.

> > I am not sure about what you mean about updating in regard to the systemd docker recommendations - is it the dnf update?
> 
> Yes, but I think we should do this change in a separate PR, it is not a direct fix to the current problem. I will try to drop it.

I am very confident that this is not really a requirement to run systemd.

File: automation/Dockerfile.fedora [hunk 4]
```diff
@@ -9,6 +9,7 @@ COPY docker_enable_systemd.sh docker_sys_config.sh ./
 RUN bash ./docker_enable_systemd.sh && rm ./docker_enable_systemd.sh
 
 RUN dnf -y install --setopt=install_weak_deps=False \
+                   systemd-udev \
```
Comment (tyll) [inline automation/Dockerfile.fedora]: Please keep the package list sorted.

Comment (EdDev) [reply automation/Dockerfile.fedora]: Done
